### ISSUE TITLE
Spider-Man 2 (U) Texture miscoloring-flickering GSdx HW renderers

### ISSUE THREAD

**Issue by bositman** [PCSX2/pcsx2#775]
Continuing from https://github.com/PCSX2/pcsx2/issues/649

PCSX2 version: 1.3.1 GIT https://github.com/PCSX2/pcsx2/commit/70990002d871bae8b2c2fdc48661453a54237638

-PCSX2 options: Defaults

-Plugins used: GSdx, SPU2-X, Lilypad, Linuz ISO/Gigaherz CDVD,rest null plugins

-Plugin settings: Defaults, tested with native, scaling, opengl depth, all accurate options, accurate date, nothing affected the issue.

-Description: When using GSdx OpenGL/D3D11 Hardware backend textures of everything rendered look garbled and flicker with different colors (distant buildings, ground you walk on, character model etc). D3D11 is a bit worse but looks like the exact same issue. Also missing the map

 OpenGL and D3D11 Software renderers both work fine (if you see stuff pop-in and out of view, it's the wide screen patch)

-How to replicate: Enter ingame and move the camera a bit

-Last known version to work: -

-PC specifications: CPU: Intel Core i5 2500K @ 4.3Ghz GPU: Gigabyte Geforce GTX660 Windforce OC 2 GB OS: Windows 7 SP1

Software renderers:
![pcsx2 2015-08-20 01-37-44-74](https://cloud.githubusercontent.com/assets/1621174/9370827/19b2332e-46dc-11e5-863c-2fdfd42a7ed2.jpg)

Hardware renderers (although the issue is much better visible when moving the camera):
![pcsx2 2015-08-20 01-37-41-37](https://cloud.githubusercontent.com/assets/1621174/9370828/19b533d0-46dc-11e5-9c46-a6838910222c.jpg)

GS dump by prafull: http://www.filedropper.com/spiderman2

**Comment by refractionpcsx2** [comment]
That looks like Mipmapping issues, do they solve themselves as you get closer?

**Comment by bositman** [comment]
That was my initial thought as well. Some do get better (probably even perfect) when you get very close, but some others that have lots of garbage, like the one at the right on the screenshot don't.

**Comment by refractionpcsx2** [comment]
I thought the ones on the right looked the worst to be honest, maybe it's just lucky bad textures which seem to fit :p

**Comment by gregory38** [comment]
I did look at a gsdump but I don't remember the outcome :( Is it better if you disable filtering?

**Comment by bositman** [comment]
Nope, no difference.

**Comment by NikolaTT** [comment]
Really sad to see this issue persisting for so long. Is it possible to help fixing it somehow (dumps etc)?

**Comment by DPersonalized** [comment]
@NikolaTT I wouldn't think it would be an easy issue to fix. There would have to be a complete rewrite of GSdx, something which I'm gonna guess @gabest11 wouldn't feel like doing.

**Comment by refractionpcsx2** [comment]
As far as I know this is mipmapping, support for which in hardware mode will be extremely difficult to pull off and there are higher priority problems to fix at the moment, before we even consider implementing it.

**Comment by FlatOutPS2** [comment]
Nope. Mipmapping off(even in SW) is actually good in this game because it makes the objects in the distance look much nicer. There are many issues in this game, including some depth issues, mipmapping isn't part of the major issues.

**Comment by refractionpcsx2** [comment]
hmm right, I always thought it was mipmapping, it looks very like it.

I presume DX11 and OGL look the same, regardless of settings?

**Comment by FlatOutPS2** [comment]
> I presume DX11 and OGL look the same, regardless of settings?

Yeah. No OGL fixes for this one.

**Comment by MrCK1** [comment]
Maybe this is similar to the corrupted squares in Guitar Hero 3; which are also not affected by mipmapping?

**Comment by ramapcsx2** [comment]
I'd still say Spiderman is some form of mipmap problem. That it doesn't look bad in software when mipmapping is off doesn't exclude this. It looks like in hardware it's pulling totally wrong textures while in software it's just always pulling the high resolution version.

**Comment by NikolaTT** [comment]
Is it possible to force the game to use the high res versions in hardware mode?

**Comment by refractionpcsx2** [comment]
The GH3 problem is a texture mapping issue I believe.  I don't think that uses mipmapping.

**Comment by gregory38** [comment]
I think both issue is that a texture got wrong data. It is more or less to the texture cache. I will an hidden option in a couple of minutes to clear an input texture to some values, so it would be easier to spot those issues.

**Comment by gregory38** [comment]
Edit: I'm not sure the option will be useful. For GH3, the game renders data into a GPU framebuffer and then read back it with an offset. Unfortunately a new texture is uploaded from the GS memory whereas the real data is inside the framebuffer.

**Comment by refractionpcsx2** [comment]
I wonder if spiderman does the same

**Comment by gregory38** [comment]
I don't remember but I think it is something close. But it could be any of the thousand texture memory trick not emulated.

**Comment by NikolaTT** [comment]
Is there a way for us to help pinpoint the problem and potentially have it fixed?

**Comment by gregory38** [comment]
Well, there are severals issues. And it is partially related to mipmapping. So it is quite hard to debug.

And main issue could be related to this kind of read back.
```
T:Oth	ID:1804	S:Info	=> TC: Lookup Target(Color) 1280x1024, miss (0x2d60, F:0x0)
T:Oth	ID:1804	S:Info	=> TC: src miss (0x2d68, F:0x13)
T:Oth	ID:1807	S:Info	=> TC: Lookup Target(Color) 1280x1024, hit: 241 (0x2d60, F:0x0)
T:Oth	ID:1807	S:Info	=> TC: dst Color hit (full): 241 (0x2d60, F:0x13)
T:Error	ID:1807	S:Mid	=> Reading RT as a packed-indexed 8 bits format
T:Error	ID:1810	S:Mid	=> ERROR: Source and Target are the same!
T:Oth	ID:1810	S:Info	=> TC: Remove Src Texture: 243 (0x2d60)
```

Here an example of texture before those draw calls (texture size is 32x32) 
![spiderman](https://cloud.githubusercontent.com/assets/1569559/16345453/3f6cb292-3a42-11e6-8bfa-513caf467796.png)

It is already a miracle that you have a kind of image for the game.

**Comment by Temflakes403** [comment]
Well, I was trying to run it and see what was happening, so I used PCSX2 1.5.0(v1.5.0-dev-2561-g8acc319a4), Direct3D 11 and a GeForce GTX 1050ti to test it. 
It would not show any image on 3x, but the audio told me it would just really slo down, might have the GPU to blame here.
On both 2x and 4x it would show image, but the textures would still look very fucked up whether Mipmapping was turned on or not, the problem was really with HW mode since SW worked perfectly, and it also slowed down, so again, might be the GPU on that front.
The anisotroic filtering was always at 16x.

**Comment by crizzy0213** [comment]
if i run it in software mode the graphics are better but the performance is slowed down, so i set it to 4 rendering threads, configured the shader settings - scanline filter, Disabled FXAA, Shade Boost disabled, and everything is running fine, just saying how it worked for me

**Comment by RedDevilus** [comment]
Tested on 1.6.0 and 1.7.0 dev 18 with default settings and with native + other upscaling methods.In HW mode u cant set the upscaling too high and the issue is still there. Buildings and other 3D models flickering like crazy, textures broken.
SW mode runs great and looks good too
HW 
![image](https://user-images.githubusercontent.com/24227051/81509605-133ee100-930c-11ea-8a66-411b1307ef6a.png)

SW
![image](https://user-images.githubusercontent.com/24227051/81509592-002c1100-930c-11ea-9132-49320c15a57e.png)
